Makefile

```diff
@@ -1,52 +1,199 @@
-DEFAULT_GOARCH := $(shell go env GOARCH)
-BUILD_HASH = $(shell git rev-parse HEAD)
-LDFLAGS += -X "github.com/mattermost/rtcd/service.buildHash=$(BUILD_HASH)"
-
-## Check go mod files consistency
-.PHONY: gomod-check
-gomod-check:
-	@echo Checking go mod files consistency
-	go mod tidy -v && git --no-pager diff --exit-code go.mod go.sum || (echo "Please run \"go mod tidy\" and commit the changes in go.mod and go.sum." && exit 1)
-
-.PHONY: check-style
-check-style: golangci-lint gomod-check
-	@echo Checking for style guide compliance
-
-.PHONY: golangci-lint
-golangci-lint: ## Run golangci-lint on codebase
-# https://stackoverflow.com/a/677212/1027058 (check if a command exists or not)
-	@if ! [ -x "$$(command -v golangci-lint)" ]; then \
-		echo "golangci-lint is not installed. Please see https://github.com/golangci/golangci-lint#install for installation instructions."; \
-		exit 1; \
-	fi; \
-
-	@echo Running golangci-lint
-	golangci-lint run ./...
 
-.PHONY: test
-test:
-	go test -v -mod=readonly -failfast -race ./...
+# Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+# See LICENSE for license information.
+
+# ====================================================================================
+# Variables
+
+# App Code location
+CONFIG_APP_CODE         += ./cmd/rtcd
+
+## Docker Variables
+# Docker executable
+DOCKER                  := $(shell which docker)
+# Dockerfile's location
+DOCKER_FILE             += ./build/Dockerfile
+# Docker options to inherit for all docker run commands
+DOCKER_OPTS             += --rm -i -u $$(id -u):$$(id -g) --platform "linux/amd64"
+# Registry to upload images
+DOCKER_REGISTRY         += docker.io
+DOCKER_REGISTRY_REPO    += mattermost
+## Docker Images
+DOCKER_IMAGE_GOLINT     += "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b"
+DOCKER_IMAGE_DOCKERLINT += "hadolint/hadolint:v2.9.2@sha256:d355bd7df747a0f124f3b5e7b21e9dafd0cb19732a276f901f0fdee243ec1f3b"
+
+## Go Variables
+# Go executable
+GO                           := $(shell which go)
+# LDFLAGS
+GO_LDFLAGS                   += -X "github.com/mattermost/${APP_NAME}/service.buildHash=$(APP_COMMIT)"
+GO_LDFLAGS                   += -X "github.com/mattermost/${APP_NAME}/service.buildVersion=$(APP_VERSION)"
+GO_LDFLAGS                   += -X "github.com/mattermost/${APP_NAME}/service.buildDate=$(BUILD_DATE)"
+# Architectures to build for
+GO_BUILD_PLATFORMS           = linux-amd64 linux-arm64 darwin-amd64 darwin-arm64
```

Comment:
I'd really like a way to override this so that a developer only has to compile for its local platform.

Reply:
Pushed https://github.com/mattermost/rtcd/pull/17/commits/1964aecde6da99cbd1c1ada56f6b011122f613a1 to assess this: 

Example for only building to `linux-amd64`
```
 GO_BUILD_PLATFORMS=linux-amd64 make go-build                       
11:35:07 [ .. ] go build rtcd-linux-amd64...
target="rtcd-linux-amd64"; \
	command="${target%%-*}"; \
	platform_ext="${target#*-}"; \
	platform="${platform_ext%.*}"; \
	export GOOS="${platform%%-*}"; \
	export GOARCH="${platform#*-}"; \
	echo export GOOS=${GOOS}; \
	echo export GOARCH=${GOARCH}; \
	CGO_ENABLED=0 \
	/opt/homebrew/bin/go build -mod=readonly -trimpath \
	-ldflags '-X "github.com/mattermost/rtcd/service.buildHash=393f0f7" -X "github.com/mattermost/rtcd/service.buildVersion=dev-393f0f7" -X "github.com/mattermost/rtcd/service.buildDate=2022-04-28 11:35"' \
	-o ./dist/rtcd-linux-amd64 \
	./cmd/rtcd || (echo `date +%H:%M:%S` [FAIL] && false)
export GOOS=linux
export GOARCH=amd64
11:35:08 [ OK ] go build rtcd-linux-amd64
```

Kudos to @pfltdv  🙇‍♂️